dvc/logger.py

```diff
@@ -58,97 +52,54 @@ class ColorFormatter(logging.Formatter):
     }
 
     def format(self, record):
+        info = record.msg
+
         if record.levelname == "INFO":
-            return record.msg
-
-        if record.levelname == "ERROR" or record.levelname == "CRITICAL":
-            exception, stack_trace = self._parse_exc(record.exc_info)
-
-            return (
-                "{color}{levelname}{nc}: {description}" "{stack_trace}\n"
-            ).format(
-                color=self.color_code.get(record.levelname, ""),
-                nc=colorama.Fore.RESET,
-                levelname=record.levelname,
-                description=self._description(record.msg, exception),
-                msg=record.msg,
-                stack_trace=stack_trace,
+            return info
+
+        if record.exc_info:
+            _, exception, _ = record.exc_info
+            cause = exception.__cause__
+
+            info = "{message}{separator}{exception}{cause}".format(
+                message=record.msg or "",
+                separator=" - " if record.msg and exception.args else "",
+                exception=exception,
+                cause=": " + str(cause) if cause else "",
             )
 
-        return "{color}{levelname}{nc}: {msg}".format(
-            color=self.color_code.get(record.levelname, ""),
-            nc=colorama.Fore.RESET,
+            if self._current_level() == logging.DEBUG:
+                trace = "".join(traceback.format_exception(*record.exc_info))
+
+                return (
+                    "{red}{levelname}{nc}: {info}\n"
+                    "{red}{line}{nc}\n"
+                    "{trace}"
+                    "{red}{line}{nc}".format(
+                        levelname=record.levelname,
+                        info=info,
+                        red=colorama.Fore.RED,
+                        line="-" * 60,
+                        trace=trace,
+                        nc=colorama.Fore.RESET,
+                    )
+                )
+
+        return "{color}{levelname}{nc}: {info}".format(
+            color=self.color_code[record.levelname],
             levelname=record.levelname,
-            msg=record.msg,
+            nc=colorama.Fore.RESET,
+            info=info,
         )
 
     def _current_level(self):
         return logging.getLogger("dvc").getEffectiveLevel()
 
-    def _is_visible(self, record):
-        return record.levelno >= self._current_level()
-
```

Comment:
Again, this is not really the point of py3 migration.

Reply:
@efiop , this is dead code, no one is using `logger._is_visible`

Reply:
Oh, good catch! Sorry, haven't realized that 😓

Reply:
No problem, should leave a message about it. :see_no_evil: